### ISSUE TITLE
Make horiztonal sorting deal with line wrapping

### ISSUE THREAD

**Issue by jgwhite** [jgwhite/ember-sortable#37]

**Comment by Cryrivers** [comment]
when is it going to be able to horizontally sort with line wrapping?

**Comment by jgwhite** [comment]
@Cryrivers I don’t currently have time to tackle this.
@igorrKurr — do you have any ideas on this one?

**Comment by igorrKurr** [comment]
@jgwhite as for me i also don't have enough time to fix it now, sorry. But if @Cryrivers can give me kind of example of desired layout maybe i will be able to make some quick solution for it.

**Comment by alexander-alvarez** [comment]
:+1: props on the great addon.

Related to this -- I was thinking of just making the sortable-group `overflow-x: auto` if the direction is 'x' so that you could reorder with things longer than the viewport? 
```
Kind of how github will add the scroll bar section around this text so that everything fits on one line
| One | Two | Three | ...                                                         ...| n | n + 1 |
```

I had to subclass `sortable-group` to get a css class that's conditional on vertical vs horizontal alignment, would have been nice if it were already there.
```
SortableGroup.extend
  classNameBindings: ['isYDirection:sortable-group-y-direction:sortable-group-x-direction']
  isYDirection: computed.equal 'direction', 'y'
```

Then again I'm still battling with the css. Is this an approach you'd consider?

**Comment by alexander-alvarez** [comment]
nvm I managed to mess around with the css to work without having to modify the sortable-group.

**Comment by jgwhite** [comment]
@alexander-alvarez thanks for your kind words.

In principle I think this is a good idea but there's nothing stopping the consuming template adding an extra class to the sortable-group or parent element.

I.e.

```hbs
{{#sortable-group direction="y" class="sortable-group-y-direction" as |group|}}
  ...
{{/sortable-group}}
```

Let me know if I've misunderstood your proposal.

Ultimately I want this addon to have the minimum possible impact on the resultant DOM. In other words: as far as possible, looking at your consuming template should tell you what will be rendered and ember-sortable should add as little extra elements and attributes as it can get away with.

**Comment by alexander-alvarez** [comment]
Yeah I get that. I guess it was more of a convenience thing since there's only ever two directions for the sortable, so that one could apply consistent CSS styling to horizontal ones vs vertical. Nonetheless I managed to get the CSS to work by wrapping it a container.

For reference in case anyone wants to take this approach until multi-line gets figured out, the parent container has `overflow:auto` and a width, and the `ul` or `ol` that gets generated by the `sortable-group` has the `whitespace:no-wrap` style applied to it, so the list can be scroll infinitely left and right as it is long.

Only outstanding thing is to get the scrollbar to scroll as you drag, will update when I address the issue.

**Comment by jgwhite** [comment]
@alexander-alvarez great points, thanks for your contribution :)

**Comment by michalsnik** [comment]
Hey, thanks for a great plugin!

I just encountered this issue and was wondering if you guys forgot about it or you won't fix this issue? It would be really great to have support for multiline lists.

**Comment by jgwhite** [comment]
@michalsnik thanks for your kind words.

I think everyone’s still keen to tackle this but the reality is that it’s a challenging problem to solve in a general way without changing the way ember-sortable works.

The fundamental tension comes from the fact that ember-sortable was designed to have the most minimal DOM-impact possible. This means we don’t depend on the DOM layout engine to compute new positions arising from a reordering. This makes it easier to tune performance but harder to adapt to arbitrary 2-dimensional layouts.

TL;DR I’d love to see a solution but personally burnt out trying to implement it in a satisfactory way.

**Comment by michalsnik** [comment]
Ahh I get it @jgwhite, I gave it a though or two but no easy solution came to my mind.

Unfortunately I don't have much time currently to figure out a proper solution in this plugin ;/ So I ended up wrapping https://github.com/RubaXa/Sortable into my ember application.

**Comment by jgwhite** [comment]
Letting this one go, for now.
